sstables/kl/reader_impl.hh

```diff
@@ -170,6 +171,9 @@ private:
     uint32_t _ttl, _expiration;
 
     bool _shadowable;
+
+    proceed_generator _gen;
+    temporary_buffer<char>* _processing_data;
```

Comment:
In the original do_process_state method we had a reference to a buffer, which we modified inside the function. We can't really feed a variable to the coroutine, and we can't have a reference member for the buffer, because there isn't really any reasonable default value for it (before we run the coroutine), so I decided to have a bare pointer (we could have an optional reference but its basically the same thing)